src/odemis/odemisd/mdupdater.py

```diff
@@ -225,41 +225,51 @@ def updatePixelDensity(unused, lens=lens, comp_affected=comp_affected, binva=bin
             binva.subscribe(updatePixelDensity)
             self._onTerminate.append((binva.unsubscribe, (updatePixelDensity,)))
 
-        # update pole position (if available), taking into account the binning
-        if model.hasVA(lens, "polePosition"):
-            def updatePolePos(unused, lens=lens, comp_affected=comp_affected):
-                # unused: because it might be polePos or binning
-
-                # the formula is: Pole = Pole_no_binning / binning
-                try:
-                    binning = comp_affected.binning.value
-                except AttributeError:
-                    binning = 1, 1
-                pole_pos = lens.polePosition.value
-                pp = (pole_pos[0] / binning[0], pole_pos[1] / binning[1])
-                md = {model.MD_AR_POLE: pp}
-                comp_affected.updateMetadata(md)
-
-            lens.polePosition.subscribe(updatePolePos, init=True)
-            self._onTerminate.append((lens.polePosition.unsubscribe, (updatePolePos,)))
-            try:
-                comp_affected.binning.subscribe(updatePolePos)
-                self._onTerminate.append((comp_affected.binning.unsubscribe, (updatePolePos,)))
-            except AttributeError:
-                pass
-
         # update metadata for VAs which can be directly copied
         for va_name, md_key in md_va_list.items():
             if model.hasVA(lens, va_name):
 
-                def updateMDFromVA(val, md_key=md_key, comp_affected=comp_affected):
+                def updateMDFromVABin(val, md_key=md_key, comp_affected=comp_affected):
                     md = {md_key: val}
                     comp_affected.updateMetadata(md)
 
                 logging.debug("Listening to VA %s.%s -> MD %s", lens.name, va_name, md_key)
                 va = getattr(lens, va_name)
-                va.subscribe(updateMDFromVA, init=True)
-                self._onTerminate.append((va.unsubscribe, (updateMDFromVA,)))
+                va.subscribe(updateMDFromVABin, init=True)
+                self._onTerminate.append((va.unsubscribe, (updateMDFromVABin,)))
+
+        # List of VA -> MD mapping when the VA values have to be divided by the binning
+        # VA name -> (MD name, binning index)
+        md_va_binning_list = {
+            "polePosition": (model.MD_AR_POLE, (0, 1)),
+            "mirrorPositionTop": (model.MD_AR_MIRROR_TOP, (1, 1)),  # Use Y for both a & b values
+            "mirrorPositionBottom": (model.MD_AR_MIRROR_BOTTOM, (1, 1)),  # Use Y for both a & b values
+        }
+
+        # update metadata for VAs which has to be divided by binning
+        for va_name, (md_key, bin_idx) in md_va_binning_list.items():
+            if model.hasVA(lens, va_name):
+                va = getattr(lens, va_name)
+
+                def updateMDFromVABin(_, va=va, md_key=md_key, bin_idx=bin_idx, comp_affected=comp_affected):
```

Comment:
Could you add a comment describing why this function is defined in this for and if loop?